post.md

```diff
@@ -0,0 +1,702 @@
+and changing state) and GDB integration. When something
+simpler is needed we may consider `mprotect()` Linux call
+for runtime code rebuilding and analysis.
+
+# How to start?
+
+Actually, this concept is based on very simple things. They
+may seem distant and abstract because nowadays they are
+typically covered with multiple layers of abstraction but
+machine code execution and organization are quite
+straightforward. I believe that knowing their principles can
+help in navigating through modern overly complex systems
+and getting a broader understanding of computing.
+
+Of course, x86 instruction set is quite broad and sometimes
+introduce very complex mechanisms. However for understanding
+how computations are divided into independent pieces and how
+do they communicate we need just to scratch the surface. Same
+with memory and process management — there is the complex theory
+behind them, but interface we get from the OS is relatively
+easy.
+
+We have CPU — computation unit and RAM for temporary data
+storage. We are usually separated from all other devices but
+we use OS interface, which reuses concepts we use for
+those two. Generally speaking, everything that happens in 
+a computer is series of passing data between components and
+transforming them in between.
+
+For example, when we display a web page we can think of it:
+
+1. we have given an address in memory.
+2. transform it into request conforming standard.
+3. pass a request to the networking device
+4. accept response
+5. transform text into an image
+6. pass the image to graphics card
+
+The job of application is just to transform the data and
+inform OS where is the product, what it is and where to pass
+it. RAM is a medium for that exchange as well as storage for
+middle products. RAM also store instructions how to perform
+those transformations. Assembly language is a textual
+representation of codes understood by CPU.
+
+# Memory
+
+We can think of RAM as a function. Every byte in memory has
+its ordinal number. We use it to read or change its value.
+Bytes are usually accessed in groups of 4(32bit) or 8(64-bit).
+
+X86 CPU never two such groups at a time, that's why it has
+own memory called registers. In modern x86 architecture, there
+are 16 64-bit general purpose registers called: RAX, RBX,
+RCX, RDX, RDI (destination index), RSI (source index),
+RBP (base pointer), RSP (stack pointer), R8-R15. Despite
+meaningful names of some, only RSP preserved special
+meaning (it was important when it was usual to code directly
+in assembly). "R" letter in beginning denote its 64-bits as
+during 40 years of x86 evolution registers grown from 16-bit
+(AX, BX, etc), to 32-bit (EAX, EBX, etc). Among special
+registers, we have RIP (instruction pointer) stores pointer
+(= ordinal number in memory) of next instruction to run and
+EFLAGS which register special situations (like zeroing
+register or overflow while adding). Those special registers
+are never accessed directly.
```

Comment:
@BartekLew this section ask for some pictures. Blog posts without any graphics are not so apparent to the potential reader.